projects/addon-editor/extensions/tiptap-node-view/component-render.ts

```diff
@@ -0,0 +1,59 @@
+import {
+    ApplicationRef,
+    ComponentFactoryResolver,
+    ComponentRef,
+    ElementRef,
+    Injector,
+    Type,
+} from '@angular/core';
+
+/**
+ * Copied from
+ * {@link https://github.com/sibiraj-s/ngx-tiptap/blob/master/projects/ngx-tiptap/src/lib/AngularRenderer.ts ngx-tiptap}
+ */
+export class TuiComponentRenderer<C, P> {
+    private applicationRef: ApplicationRef;
+    private componentRef: ComponentRef<C>;
+
+    constructor(component: Type<C>, injector: Injector, props: Partial<P>) {
+        this.applicationRef = injector.get(ApplicationRef);
+
+        const componentFactoryResolver = injector.get(ComponentFactoryResolver);
+        const factory = componentFactoryResolver.resolveComponentFactory(component);
+
+        this.componentRef = factory.create(injector, []);
+
+        // set input props to the component
+        this.updateProps(props);
+
+        // Attach to the view so that the change detector knows to run
+        this.applicationRef.attachView(this.componentRef.hostView);
+    }
+
+    get instance(): C {
+        return this.componentRef.instance;
+    }
+
+    get elementRef(): ElementRef {
+        return this.componentRef.injector.get(ElementRef);
+    }
+
+    get dom(): HTMLElement {
+        return this.elementRef.nativeElement;
+    }
+
+    updateProps<T extends P>(props: Partial<T>): void {
+        Object.entries(props).forEach(([key, value]) => {
+            this.instance[key as keyof C] = value as C[keyof C];
+        });
+    }
+
+    detectChanges(): void {
+        this.componentRef.changeDetectorRef.detectChanges();
+    }
+
+    destroy(): void {
+        this.componentRef.destroy();
+        this.applicationRef.detachView(this.componentRef.hostView);
```

Comment:
This looks like it's unnecessary:
https://angular.io/api/core/ApplicationRef#attachView
You can test it to be sure, try attaching it after `destroy` — if it's still attached at that point, it would throw. If this is indeed unnecessary, there's no need to store `applicationRef` as class property.

Reply:
i removed this line